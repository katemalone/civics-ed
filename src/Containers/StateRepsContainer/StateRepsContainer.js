import React from 'react'; 
import { connect } from 'react-redux';
import StateReps from '../StateReps/StateReps'

export const StateRepsContainer = ({ stateRepsInfo }) => {
  const display = stateRepsInfo.map(rep => {
    return (
      <StateReps
        key={rep.full_name}
        name={rep.full_name}
        image={rep.photo_url}
        email={rep.email}
        party={rep.party}
        chamber={rep.chamber}
      />
    )
  })
  return( 
    <div>
      {display}
    </div>
  )
}

export const mapStateToProps = ({ stateRepsInfo }) => ({
  stateRepsInfo
})


export default connect(mapStateToProps)(StateRepsContainer)