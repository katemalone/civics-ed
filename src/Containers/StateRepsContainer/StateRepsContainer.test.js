import React from 'react';
import { shallow } from 'enzyme';
import { StateRepsContainer } from './StateRepsContainer';

describe('StateRepsContainer', () => {

  let mockRepInfo =
    [{
      "id": "AKL000016",
      "leg_id": "AKL000016",
      "all_ids": ["AKL000016", "AKL000089", "AKL000165"],
      "full_name": "Bert Stedman",
      "first_name": "Bert",
      "last_name": "Stedman",
      "suffix": "",
      "photo_url": "http://akleg.gov/images/legislators/smn.jpg"
    }];


  it('should match snapShot', () => {
    const wrapper = shallow(
      <StateRepsContainer stateRepsInfo={mockRepInfo} />);
    expect(wrapper).toMatchSnapshot()
  });
});