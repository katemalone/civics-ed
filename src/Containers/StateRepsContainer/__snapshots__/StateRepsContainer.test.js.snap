// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateRepsContainer should match snapShot 1`] = `
<div>
  <StateReps
    image="http://akleg.gov/images/legislators/smn.jpg"
    key="Bert Stedman"
    name="Bert Stedman"
  />
</div>
`;
